tool/tsh/resolve_default_addr.go

```diff
@@ -0,0 +1,152 @@
+/*
+Copyright 2021 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"strconv"
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+type raceResult struct {
+	addr string
+	err  error
+}
+
+// raceRequest drives an HTTP request to completion and posts the results back
+// to the supplied channel.
+func raceRequest(ctx context.Context, cli *http.Client, addr string, results chan<- raceResult) {
+	target := fmt.Sprintf("https://%s/", addr)
```

Comment:
What if some other webserver is listening on the default https port, could we mistakenly assume that's the teleport proxy?

I'm imagining a setup with teleport listening on `example.com:3080` and another webserver on `example.com:443`, and the webserver responds first.

Reply:
Not seeing any Server string w ecould key off.

Reply:
One idea would be GETing `/webapi/ping` (or even just using `webclient.Ping`) to be reasonably sure we're talking to a teleport proxy. Not a huge concern because the user can just specify the port, like you say, but I'd hate for this to trip someone up when it's supposed to make things simpler

Reply:
Yeah I think /webapi/ping is a good endpoint to hit to determine whether we're talking to a proxy.

Reply:
Done